x-pack/plugins/apm/public/components/alerting/ui_components/apm_rule_group_by.tsx

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiComboBox } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React, { useCallback } from 'react';
+import { TRANSACTION_NAME } from '../../../../common/es_fields/apm';
+
+interface Props {
+  options: { groupBy: string[] | string | undefined };
+  preSelectedOptions: string[];
+  onChange: (groupBy: string | null | string[]) => void;
+  errorOptions?: string[];
+}
+
+const fields: string[] = [TRANSACTION_NAME];
+
+export function APMRuleGroupBy({
+  options,
+  preSelectedOptions,
+  onChange,
+  errorOptions,
+}: Props) {
+  const handleChange = useCallback(
+    (selectedOptions: Array<{ label: string }>) => {
+      const groupByOption = selectedOptions.map((option) => option.label);
+      onChange([...new Set(preSelectedOptions.concat(groupByOption))]);
+    },
+    [onChange, preSelectedOptions]
+  );
+
+  const getPreSelectedOptions = () => {
+    return preSelectedOptions.map((field) => ({
+      label: field,
+      color: 'lightgray',
+      disabled: true,
+    }));
+  };
```

Comment:
I would rather keep it as function only because it's naming matches with the other function in below where it is called - 

```
const selectedOptions = [
    ...getPreSelectedOptions(),
    ...getUserSelectedOptions(options.groupBy),
  ];
```